addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -2059,6 +2059,11 @@ export class PosStore extends WithLazyGetterTrap {
             }
         );
     }
+    orderContainsProduct(product) {
+        return this.getOrder()?.lines.some(
+            (line) => line.product_id?.product_tmpl_id.id == product.id
+        );
```

Comment:
```suggestion
        const lines = this.getOpenOrders().flatMap((o) => o.lines);
        return lines.some((l) => l.product_id.product_tmpl_id.id === product.id)
```

We must check in all open orders